packages/utils/src/types/publicKey.ts

```diff
@@ -0,0 +1,142 @@
+import { publicKeyConvert, publicKeyCreate, ecdsaRecover } from 'secp256k1'
+import { u8aToHex, u8aEquals, stringToU8a } from '../u8a'
+import PeerId from 'peer-id'
+import { pubKeyToPeerId } from '../libp2p'
+import { Address, Hash } from './primitives'
+
+export class PublicKey {
+  // Cache expensive computation result
+  private _address: Address
+
+  private constructor(private arr: Uint8Array) {}
```

Comment:
For some reason I had the impression that we will be caching both values - compressed and uncompressed (one of which gets created ad-hoc when needed and cached since then). But it seems that the value is converted back and forth while the `arr` is being overwritten with one key form or another?

Reply:
Only the uncompressed version because compressing is cheap. Should make it also easier to read.

Reply:
> But it seems that the value is converted back and forth while the arr is being overwritten with one key form or another?

Yes, it is. But only the uncompressed representation is kept in memory